assets/src/edit-story/components/canvas/page.js

```diff
@@ -50,37 +51,42 @@ function Page() {
 		}
 		evt.stopPropagation();
 
+		// Persist the event so that a dragstart could be
+		// later triggered without having to select the element first.
 		if ( 'pointerdown' === evt.type ) {
 			evt.persist();
 			setPushEvent( evt );
 		}
 	}, [ toggleElementIdInSelection, selectElementById ] );
 
-	const selectedElement = selectedElements.length === 1 ? selectedElements[ 0 ] : null;
+	const singleSelection = 1 === selectedElements.length;
+	const hasSelection = 1 <= selectedElements.length;
 
 	return (
 		<Background>
 			{ currentPage && currentPage.elements.map( ( { id, ...rest } ) => {
-				const isSelected = Boolean( selectedElement && selectedElement.id === id );
-
+				const isSelected = Boolean( selectedElements.filter( ( { id: selectedId } ) => id === selectedId ).length );
 				return (
 					<Element
 						key={ id }
 						setNodeForElement={ setNodeForElement }
+						setTargetRefs={ setTargetRefs }
```

Comment:
FI: `setNodeForElement` and `setTargetRefs` are quite similar here, however, just used in different places. Related to what @dvoytenko mentioned then ideally we'd have one place to store the mapped refs to elements. Still looking into this